main/ZgatewayBT.ino

```diff
@@ -341,8 +341,10 @@ class MyAdvertisedDeviceCallbacks : public BLEAdvertisedDeviceCallbacks
     { //if not black listed mac we go AND if we have no white mac or this mac is  white we go out
       if (advertisedDevice.haveName())
         BLEdata.set("name", (char *)advertisedDevice.getName().c_str());
+      #if pubBLEManufacturerData
       if (advertisedDevice.haveManufacturerData())
         BLEdata.set("manufacturerdata", (char *)advertisedDevice.getManufacturerData().c_str());
+      #endif
```

Comment:
I'm using the #if directive in ZgatewayBT.ino instead of the #ifndef directive so the user can override the value of 'pubBLEManufacturerData' in the platformio.ini file. The value defaults to 'false' if no define is made and the ManufacturerData is not published.

If using the #ifdef / #ifndef directives, it can be confusing to the user to set their value as true/false if not needed. They both evaluate to the value being defined and the code running.

Reply:
I've just learnt something, I didn't know that we can use #if 
Perfect!